src/analysis/special_functions/bernstein.lean

```diff
@@ -0,0 +1,301 @@
+begin
+  have h' : (n : ℝ) ≠ 0 := (ne_of_lt h).symm,
+  apply_fun (λ x : ℝ, x * n) using group_with_zero.mul_right_injective h',
+  apply_fun (λ x : ℝ, x * n) using group_with_zero.mul_right_injective h',
+  dsimp,
+  conv_lhs { simp only [finset.sum_mul, z], },
+  conv_rhs { rw div_mul_cancel _ h', },
+  have := bernstein_polynomial.variance ℝ n,
+  apply_fun (λ p, polynomial.aeval (x : ℝ) p) at this,
+  simp [alg_hom.map_sum, finset.sum_range, ←polynomial.nat_cast_mul] at this,
+  convert this using 1,
+  { congr' 1, funext k,
+    rw [mul_comm _ (n : ℝ), mul_comm _ (n : ℝ), ←mul_assoc, ←mul_assoc],
+    congr' 1,
+    field_simp [h],
+    ring, },
+  { ring, },
+end
+
+end bernstein
+
+open bernstein
+
+local postfix `/ₙ`:2000 := z
+local notation `|`x`|` := abs x
+
+/--
+The `n`-th approximation of a continuous function on `[0,1]` by Bernstein polynomials,
+given by `∑ k, f (k/n) * bernstein n k x`.
+-/
+def bernstein_approximation (n : ℕ) (f : C(I, ℝ)) : C(I, ℝ) :=
+∑ k : fin (n+1), f k/ₙ • bernstein n k
+
+/-!
+We now set up some of the basic machinery of the proof that the Bernstein approximations
+converge uniformly.
+
+A key player is the set `S f ε h n x`,
+for some function `f : C(I, ℝ)`, `h : 0 < ε`, `n : ℕ` and `x : I`.
+This is the set of points `k` in `fin (n+1)` such that `f(k/n)` is within `ε/2` of `f x`.
+
+We show that if `k ∉ S`, then `1 ≤ δ^-2 * (x - k/n)^2`.
+-/
+
+namespace bernstein_approximation
+
+@[simp] lemma apply (n : ℕ) (f : C(I, ℝ)) (x : I) :
+  bernstein_approximation n f x = ∑ k : fin (n+1), f k/ₙ * bernstein n k x :=
+by simp [bernstein_approximation]
+
+/--
+The set of points `k` so `k/n` is within `δ` of `x`.
+-/
+def S (f : C(I, ℝ)) (ε : ℝ) (h : 0 < ε) (n : ℕ) (x : I) : finset (fin (n+1)) :=
+{ k : fin (n+1) | dist k/ₙ x < f.modulus (ε/2) (half_pos h) }.to_finset
+
+/--
+If `k ∈ S`, then `f(k/n)` is close to `f x`.
+-/
+lemma lt_of_mem_S
+  {f : C(I, ℝ)} {ε : ℝ} {h : 0 < ε} {n : ℕ} {x : I} {k : fin (n+1)} (m : k ∈ S f ε h n x) :
+  |f k/ₙ - f x| < ε/2 :=
+begin
+  apply f.dist_lt_of_dist_lt_modulus (ε/2) (half_pos h),
+  simpa [S] using m,
+end
+
+/--
+If `k ∉ S`, then as `δ ≤ |x - k/n|`, we have the inequality `1 ≤ δ^-2 * (x - k/n)^2`.
+This particular formulation will be helpful later.
+-/
+lemma le_of_mem_S_compl
+  {f : C(I, ℝ)} {ε : ℝ} {h : 0 < ε} {n : ℕ} {x : I} {k : fin (n+1)} (m : k ∈ (S f ε h n x)ᶜ) :
+  (1 : ℝ) ≤ (f.modulus (ε/2) (half_pos h))^(-2 : ℤ) * (x - k/ₙ) ^ 2 :=
+begin
+  simp only [finset.mem_compl, not_lt, set.mem_to_finset, set.mem_set_of_eq, S] at m,
+  field_simp,
+  erw [le_div_iff (pow_pos f.modulus_pos 2), one_mul],
+  apply sqr_le_sqr,
+  rw abs_eq_self.mpr (le_of_lt f.modulus_pos),
+  rw [dist_comm] at m,
+  exact m,
+end
+
+end bernstein_approximation
+
+open bernstein_approximation
+open bounded_continuous_function
+open filter
+
+open_locale topological_space
+
+/--
+This is the proof given in [Richard Beals' *Analysis, an introduction*][beals-analysis], §7D,
```

Comment:
The docstring should first say what the theorem is proving (although having the reference you give in the docstring is also a very good idea, but it should come second).